xbmc/windowing/win10/WinSystemWin10.cpp

```diff
@@ -650,11 +651,41 @@ bool CWinSystemWin10::MessagePump()
   return m_winEvents->MessagePump();
 }
 
-int CWinSystemWin10::GetGuiSdrPeakLuminance() const
+/*!
+ * \brief Max luminance for GUI SDR content in HDR mode.
+ * \return Max luminance in nits, lower than 10000.
+*/
+float CWinSystemWin10::GetGuiSdrPeakLuminance() const
 {
   const auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
 
-  return settings->GetInt(CSettings::SETTING_VIDEOSCREEN_GUISDRPEAKLUMINANCE);
+  // use cached system value as this is called for each frame
+  if (settings->GetBool(CSettings::SETTING_VIDEOSCREEN_USESYSTEMSDRPEAKLUMINANCE) &&
+      m_validSystemSdrPeakLuminance)
+    return m_systemSdrPeakLuminance;
+
+  // Max nits for 100% UI setting = 1000 nits, < 10000 nits, min 80 nits for 0%
+  int guiSdrPeak = settings->GetInt(CSettings::SETTING_VIDEOSCREEN_GUISDRPEAKLUMINANCE);
+  return 10000.0f / ((100 - guiSdrPeak) * 1.15f + 10);
+}
+
+/*!
+ * \brief Test support of the OS for a SDR max luminance in HDR mode setting
+ * \return true when the OS supports that setting, false otherwise
+*/
+bool CWinSystemWin10::HasSystemSdrPeakLuminance()
```

Comment:
```suggestion
bool CWinSystemWin10::HasSystemSdrPeakLuminance() const
```

Reply:
Not possible without further changes in CWinSystemWin32. CWinSystemBase::HasSystemSdrPeakLuminance is parent for both

Can't be const in CWinSystemWin32::HasSystemSdrPeakLuminance because of GetDisplayDetails(m_hMonitor)
could be made const with additional changes in GetDisplayDetail.

Since there is a cached value now I can reinstate the const of GetGuiSdrPeakLuminance BTW (used to depend on GetDisplayDetails(m_hMonitor) as well but not anymore)